### PR TITLE
The HTTP/3 protocol on iroh

### PR DESCRIPTION
**Pull request by flub**
This implements a low-level API from hyper which can be used directly.  It also includes primitive support for running an axum server.

The latter does some extra copies and puts too much in memory.  It should still be generalised into more generic hyper support with parametrised body etc.

This currently relies on a git reference to iroh-net.  The 0.28 iroh-net release should fix this, probably best to wait until that release to merge this so that this does not break CI for the other examples randomly as there's no Cargo.lock checked in.

### CODE REVIEW

**Comment by matheus23** [comment]
Should we add the `h3::quic` trait impls to iroh-net under a feature flag?

**Comment by flub** [comment]
> Should we add the `h3::quic` trait impls to iroh-net under a feature flag?

oh, i hadn't considered that.  I think it might be too early.  I don't think those traits are 1.0 yet and putting them in an unstable feature flag is a bit meh and sets too much precedence for throwing things into iroh-net.  Should be fine as a separate crate for now.

But if we want to take this more serious we should indeed give it its own repo and some docs etc.

**Comment by dignifiedquire** [comment]
@flub do we want to merge this?

**Comment by flub** [comment]
> @flub do we want to merge this?

totally!

File: h3-iroh/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+axum = { version = "0.7", optional = true }
+bytes = "1"
+futures = "0.3"
+h3 = { version = "0.0.6", features = ["tracing"] }
+http = { version =  "1.1", optional = true }
+http-body = { version = "1", optional = true }
+http-body-util = { version = "0.1", optional = true }
+hyper = { version = "1.5", optional = true }
+hyper-util = { version = "0.1", optional = true }
+iroh-net = { git = "http://github.com/n0-computer/iroh/", branch = "main" }
```
Comment (dignifiedquire) [inline h3-iroh/Cargo.toml:18]: @flub I think this won't work anymore 😅

Comment (flub) [reply h3-iroh/Cargo.toml:18]: oops, yes: https://github.com/n0-computer/iroh-experiments/pull/21

**Pull request merged**